ims/einstein/iscsi/tgt.py

```diff
@@ -11,180 +12,154 @@
 
 
 class TGT(ISCSI):
-    '''
-        Class for implementing TGT
-    '''
+    """ Class for implementing TGT """
 
-    # We should get all the config related to ceph in here.
-    # Also, root_password is something that can be avoided.
-
-    # Also removed sudo everywhere since it was causing issues
-    def __init__(self, fs_config_loc, fs_user, root_password):
-        self.arglist = ["service", "tgtd"]
+    def __init__(self, fs_config_loc, fs_user, fs_pool):
         self.TGT_ISCSI_CONFIG = "/etc/tgt/conf.d/"
         self.fs_config_loc = fs_config_loc
         self.fs_user = fs_user
-        self.root_password = root_password
+        self.fs_pool = fs_pool
 
     @log
     def start_server(self):
-        '''
-        Have to parse the output and send a status code.
-        :return:
-        '''
-        # arglist = self.arglist
-        # arglist.append("start")
-        # Need to fix shell=True everywhere
-        command = "service tgtd start"
-        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        output, err = proc.communicate(self.root_password +
-                                       "\n")
-        logger.debug("Output = %s, Error = %s", output, err)
-        if self.show_status() is 'Running':
+        """
+        Starts the tgt service
+
+        :return: None
+        """
+        try:
+            command = "service tgtd start"
```

Comment:
@knikolla IMO, start_server & stop_server are functions specific to an iscsi server and I think, it makes sense to have them in that particular file rather than in shell.py. To me, shell.py is a util file that each driver could have. If there is an iscsi server that needs to be started from python or some other language, having it in its own set of scripts, having a generic one may not work either. 

I understand your point of abstracting shell code away but I think this is cleaner we don't know, what iscsi server uses what. Let me know what you think.